### DISCUSSION TITLE
Welcome to pikaraoke Discussions!

### DISCUSSION THREAD

**Discussion by vicwomg** [vicwomg/pikaraoke#187] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to ask for technical support and connect with other members of our community. Many general questions, features requests, and troubleshooting discussions have been occurring in the issues tracker, which probably is not the best place for them. 

In the Discussions se iont, we hope that you:
  * Ask questions.
  * Request technical support
  * Share feature ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we build together 💪.

 To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by Hartebee5t** [comment]
Great idea, opening a discussion forum.
I'm David, I live in Belfast, Northern Ireland and I've been running PiKaraoke on Pi 4B since Christmas 2019.

**Reply by vicwomg** [reply]
Glad to put a name to the Avatar David! Thanks for being such a great contributor for all these years.

**Comment by vicwomg** [comment]
Vic here, original author or pikaraoke. I'm based in San Francisco and in addition to coding I'm a professional jazz guitarist. Like most projects of mine, I started with a personal need and not being able to find a good solution out there. I also love learning new things. 

I hear from a lot of folks using this project, saying things "Sorry for the dumb question but I'm new to git / coding / linux / raspberry pi" and I just want to say that you're all very welcome to ask anything here and my hope is that I or other users can help without making you feel like you're asking "dumb questions". We all start somewhere.

**Comment by giggaa** [comment]
Hi. I'm Tran from New Orleans. I have some coding experience but not in Python or web stuff. I looked over the code and it looks like the code is sending request to VLC through http and the command. I was wondering if you can point me in the right direction to code in the option to use audio tracks.

Right now, I'm converting the files to use audio track 2, which is the non vocal track. Is there a way to implement an option to choose audio track 1 (with vocal) and audio track 2 (without vocal) and send it to VLC? Any help is greatly appreciated. Thanks.

**Reply by vicwomg** [reply]
If you can track down the vlc command line option to specify the audio track then it should be pretty straightforward to add to the launch command

**Reply by giggaa** [reply]
I was able to implement the changing of the audio track. The VLC command is audio_track&val=1, for audio track 1. Now, is there a way to determine how many audio tracks a file has? If I send in a command to use audio track 2 but the file only has 1 audio track then it will error out. Thanks for the reply.

**Reply by Theon007** [reply]
can we add this to the program? or could you share how it's done? thanks! i am thinking having a button command to switch between tracks if the users wants to, just like the pitch adjustment.

**Comment by nightsyde** [comment]
Is there a way to direct the program to look at an external drive for files? I have almost a T of files, and am working on moving to a system a little more portable than my laptop. I have not taken the time to peruse the code to see the search algorithms yet.

**Reply by vicwomg** [reply]
Yes, you need to launch the program with the '-d (directory name)' option and specify your USB drive directory

**Reply by nightsyde** [reply]
awesome! Took a while, but most of the files are available. Now, to play around with it and see what all it can do, and what ells and whistles I need.

**Comment by s177oke77** [comment]
Thank you for such a great project. My family has used this on a RPI as a karaoke machine and love how everything is setup. I was wondering how I would go about tweaking the system by adding in the function to intercept keypress so that we can skip or restart a song with a keypress from a keyboard (we will end up using Flirc with a remote). I am just unsure where to add the: 
if event.type == pygame.KEYDOWN:
            if event.key== pygame.K_w:
and what function I would call to skip/restart. Thank you again for all the time you have put into this!

**Comment by NguyenDuong-AU** [comment]
Hi, I'm based in Australia and only found PK last year, before that I was using musicxmatch browser plugin with Chrome but that wasn't a great solution and didn't has as much flexibility as this. 

Just wondering if anyone has modified the search.html page to show results more similar to a youtube layout? So instead of a dropdown menu with each song artist and title, have a section on the search page showing the preview thumbnail against each result? Just makes it easier to identify when there are a lot of songs with similar titles but you're trying to find a specific one:

So currently like this:
![image](https://user-images.githubusercontent.com/7434489/166133927-ae45474a-671a-43f5-9657-452a97d154f4.png)

Was hoping to modify it to show something like what youtube shows like this so I can see a preview of each result and be able to identify them without clicking through each one, one at a time:
![image](https://user-images.githubusercontent.com/7434489/166133942-69386c80-a11d-42ca-9466-9d25eb0785a4.png)

**Comment by thiago-tobaldini** [comment]
Great projetc, is there a way to put a score at the end of the song?

Even if it's a random score, it seems that people get more excited when there's something like that, could you guide me to include this in the project?

**Comment by BrandtVavasour** [comment]
@vicwomg Awesome project. Been using this for a couple of months now and loving it. The way you've built the app has inspired me to build a flutter app that utilizes the API calls on the pikaraoke webserver. I'll release the source code when it's up and working for others to use. The flutter app should allow anyone to run on any platform they choose, phone, windows, linux, macos or the browser as now. Here is a sneak peak of the mobile version I'm building. 
![image](https://user-images.githubusercontent.com/9376621/168585680-e7652d2c-6877-43ff-860f-9b8acebc58f2.png)



Keep those APIs coming so I can continue to pack the app with the features you're building.

**Reply by vicwomg** [reply]
That's awesome! Let me know when you have something testable and I'd love to check it out.

**Reply by BrandtVavasour** [reply]
I've been getting more time to work on this. I've had to makes some slight adjustments to the python app to get some things working. Once I think i've finalised what I think is needed to have the app run (CORS!) i'll create a pull request.

**Comment by Helvio88** [comment]
Hey there! I have a complimentary library that I called "karaokeer", check out helv-io/karaokeer - it uses AI to create karaoke lyrics on the fly in the ASS file format (see libass) which should be easy to add to a standard HTML player and is natively supported by VLC.

Basically you search for a song, powered by Genius API, then the lyrics are downloaded and aligned with the voice via AI. It also separates the voice and instruments. Everything is provided as a MP4 file with 3 audio tracks to choose from.

Let me know if you want to work together on joining those projects!

helvio (number eighty eight) at gmail
